src/derivatives/broadcast.jl

```diff
@@ -264,8 +264,10 @@ end
 
 @inline _materialize(f, args) = broadcast(f, args...)
 
-for (M, f, arity) in DiffRules.diffrules()
-    isdefined(ReverseDiff, M) || continue
+for (M, f, arity) in DiffRules.diffrules(; filter_modules=nothing)
+    if !(isdefined(@__MODULE__, M) && isdefined(getfield(@__MODULE__, M), f))
```

Comment:
Do we know which functions will not be defined here?

Reply:
None currently. The problem is that this check was missing and therefore introducing new rules (such as the ones for LogExpFunctions) broke ReverseDiff (and ForwardDiff and Tracker...). Therefore we had to add the `filter_modules` keyword argument and set its default value to the list of the previous list of packages (i.e., SpecialFunctions and NaNMath). This is the fix for the initial problem and will avoid such issues in the future, if rules for other packages are added to DiffRules.

Reply:
I see. It might be useful to know if there are rules added to DiffRules and missed by ReverseDiff, so maybe turn this to a warning instead?

Reply:
OK, I added a warning in the package and an error in the tests (to ensure that all available rules are tested but to avoid breaking the package if DiffRules supports additional packages that are not loaded).